mistakedetection/src/ca/mcgill/sel/mistakedetection/MistakeDetection.java

```diff
@@ -3304,10 +3320,42 @@ public static String taggerOut(String taggerInput) {
     return maxentTagger.tagString(taggerInput);
   }
 
-  private static boolean spellingMistakeCheck(String name1, String name2) {
-    int lDistance = levenshteinDistance(name1, name2);
-    return lDistance > 0 && lDistance <= MAX_LEVENSHTEIN_DISTANCE_ALLOWED;
+  private static boolean isSpelledIncorrectly(int lDistance, int studStringLength, int instStringLength) {
+ 
+    if( lDistance == 0) {
+      return false;
+    }
+    int limit = 2;
+    if (instStringLength > 5 || studStringLength > 5) {
+      return lDistance <= limit;
+    } else if (instStringLength > 2 || studStringLength > 2) {
```

Comment:
Use CONSTANTs instead of numbers like 2 and 5 so they can updated in one place if they ever need to change in the future, eg, to 3 and 6 🧙